code/modules/mob/living/living_defense.dm

```diff
@@ -1,44 +1,34 @@
 
 /*
-	run_armor_check(a,b)
+	run_armor_check(a,b,c,d,e,f,g)
 	args
 	a:def_zone - What part is getting hit, if null will check entire body
 	b:attack_flag - What type of attack, bullet, laser, energy, melee
+	c: What text is to be shown should the blow be fully negated
+	d: What text is to be shown should the blow be partially negated
+	e: Modifier on top of the armor, multiplier
+	f: Whether messages should be shown or not to the user
+	g: Armor penetration - How much armor to be negated.
 
 	Returns
-	0 - no block
-	1 - halfblock
-	2 - fullblock
+	percent damage reduction
 */
 /mob/living/proc/run_armor_check(var/def_zone = null, var/attack_flag = "melee", var/absorb_text = null, var/soften_text = null, modifier = 1, var/quiet = 0, var/armor_penetration = 0)
-	var/armor = getarmor(def_zone, attack_flag)
-	var/absorb = 0
+	var/armor = (getarmor(def_zone, attack_flag)-armor_penetration)*modifier
 
-	if(prob(armor * modifier))
-		absorb += 1
-	if(prob(armor * modifier))
-		absorb += 1
-
-	if(prob(armor_penetration))
-		absorb -= 1
-	if(prob(armor_penetration))
-		absorb -= 1
-
-	if(absorb >= 2)
+	if(armor >= 100) //Absolutely no damage
 		if(!quiet)
 			if(absorb_text)
 				show_message("[absorb_text]")
 			else
 				show_message("<span class='borange'>Your armor ABSORBS the blow!</span>")
-		return 2
-	if(absorb == 1)
+	else if(armor > 0)
```

Comment:
Should tweak the combat message if this is going to be shown every time, otherwise people will be shown the orange "softens the blow" text every time. Maybe increase the threshold instead to 50 and above?

Reply:
that sounds fair

Reply:
I think this message is no longer needed